graphqlapi/src/main/java/io/stargate/graphql/web/resources/cqlfirst/GraphqlCache.java

```diff
@@ -81,10 +80,10 @@ public GraphQL getDdl() {
     return ddlGraphql;
   }
 
-  public GraphQL getDml(String keyspace, Map<String, String> headers) {
-    String decoratedName = persistence.decorateKeyspaceName(keyspace, headers);
-    DmlGraphqlReference ref = dmlGraphqls.get(decoratedName);
-    return ref == null ? null : ref.get();
+  public GraphQL getDml(String keyspaceName, Map<String, String> headers) {
+    String decoratedKeyspaceName = persistence.decorateKeyspaceName(keyspaceName, headers);
+    return dmlGraphqls.computeIfAbsent(
+        decoratedKeyspaceName, __ -> buildDmlGraphql(keyspaceName, headers));
   }
 
   public GraphQL getDefaultDml() {
```

Comment:
Would it make sense to pass headers to `getDefaultDml()` as well? They seem to be available in the context of calls to this method.

Reply:
Yes, I'll add it.

It turns out that the "default keyspace" feature is disabled in the context where decorated keyspaces are a thing (see `DISABLE_DEFAULT_KEYSPACE`), but it doesn't hurt to be consistent.